lib/internal/Magento/Framework/View/Layout/ProcessorInterface.php

```diff
@@ -126,7 +126,15 @@ public function getFileLayoutUpdatesXml();
     public function getContainers();
 
     /**
-     * Return cache ID based current area/package/theme/store and handles
+     * Add cache key for generating different cache id for same handles
+     *
+     * @param array|string $cacheKey
+     * @return ProcessorInterface
+     */
+    public function addCacheKey($cacheKey);
```

Comment:
To maximize backwards compatibility with the existing customizations and extensions we can not introduce a new public method in the interface. More information is available in the [Backwards Compatible Development Guide](http://devdocs.magento.com/guides/v2.1/ext-best-practices/extension-coding/backwards-compatible-development/index.html)